app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -206,17 +207,184 @@ class StateFragmentTest {
+    }
+  }
+
   //  5. Configuration change after submitting the wrong answer to show that the old answer & re-ask
   //     of the question stay the same.
+  @Test
+  fun testStateFragment_loadExp_submitWrongAnswer_changeConfiguration_feedbackAndQuestionVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/3")
+      clickSubmitAnswerButton()
+
+      rotateToLandscape()
+
+      scrollToViewType(FEEDBACK)
+      onView(withId(R.id.feedback_text_view)).check(matches(isDisplayed()))
+      onView(withId(R.id.fraction_input_interaction_view)).check(matches(isDisplayed()))
+    }
+  }
+
   //  6. Backward/forward navigation along with configuration changes to verify that you stay on the
   //     navigated state.
+  @Test
+  fun testStateFragment_loadExp_moveForward_changeConfiguration_stayOnTheNavigatedState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "Which bird can sustain flight for long periods of time?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_moveForward_moveBackward_changeConfiguration_stayOnSameState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
   //  7. Verifying that old answers were present when navigation backward/forward.
+  @Test
+  fun testStateFragment_loadExp_secondState_submitAnswer_moveForward_moveBackward_answerVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+
+      scrollToViewType(SUBMITTED_ANSWER)
+      onView(withId(R.id.submitted_answer_text_view)).check(
+        matches(
+          withText("1/2")
+        )
+      )
+    }
+  }
+
   //  8. Testing providing the wrong answer and showing hints.
+  @Test
+  @Ignore("Failing")
```

Comment:
This will be helpful if you mention more why this is failing. 
Adding @FareesHussain for Hint and Solution context.

Reply:
Hints aren't visible while testing in FRACTIONS_EXPLORATION_ID_1, do we lack necessary framework for that while testing?

Reply:
@khushal1707 You can test this in the 4th exploration.

https://github.com/oppia/oppia-android/issues/2476 Refer this to follow steps to the exploration with multiple hints.

Reply:
Okay, I'll take reference from that